scalajslib/src/ScalaJSModule.scala

```diff
@@ -197,8 +197,22 @@ trait ScalaJSModule extends scalalib.ScalaModule { outer =>
 
   def moduleKind: T[ModuleKind] = T { ModuleKind.NoModule }
 
-  def useECMAScript2015: T[Boolean] = T { 
-    !scalaJSVersion().startsWith("0.")
+  @deprecated("Use esFeatures().esVersion instead", since = "mill after 0.10.0-M5")
+  def useECMAScript2015: T[Boolean] = T {
+    esFeatures().esVersion != ESVersion.ES5_1
+  }
+
+  def esFeatures: T[ESFeatures] = T {
+    if (useECMAScript2015.ctx.enclosing != s"${classOf[ScalaJSModule].getName}#useECMAScript2015") {
+      Result.Failure(
```

Comment:
Is it really necessary to fail the build? Can't we just use either ES5_1 or ES2015 when `useECMAScript2015` was overridden? I think the big red warning might be annoying enough on the CLI, and in the editor the deprecation warning also indicates the outdated target. It is far more important to make transition to newest Mill as easy as possible. The less likely older Mill versions are used, the better.

Reply:
You are right! I couldn't see how to make it work.
But actually the new defaults are all expressible using the old ones.